shell/common/node_bindings.cc

```diff
@@ -122,6 +123,60 @@ void stop_and_close_uv_loop(uv_loop_t* loop) {
 
 bool g_is_initialized = false;
 
+bool IsPackagedApp() {
+  base::FilePath exe_path;
+  base::PathService::Get(base::FILE_EXE, &exe_path);
+  base::FilePath::StringType path = base::ToLowerASCII(exe_path.value());
+
+#if defined(OS_WIN)
+  return path == FILE_PATH_LITERAL("electron.exe");
+#else
+  return path == FILE_PATH_LITERAL("electron");
+#endif
+}
+
+// Initialize NODE_OPTIONS to pass to Node.js
+void SetNodeOptions(base::Environment* env) {
+  // Options that are unilaterally disallowed
+  const std::set<std::string> disallowed = {
+      "--openssl-config", "--use-bundled-ca", "--use-openssl-ca",
+      "--force-fips", "--enable-fips"};
+
+  // Subset of options allowed in packaged apps
+  const std::set<std::string> allowed_in_packaged = {"--max-http-header-size"};
+
+  if (env->HasVar("NODE_OPTIONS")) {
+    std::string options;
+    env->GetVar("NODE_OPTIONS", &options);
+    std::vector<std::string> parts = base::SplitString(
+        options, " ", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+
+    std::string new_options = options;
```

Comment:
`options` doesn't seem to be used anywhere other than for copying here, can instead use it directly and avoid the copy.

Reply:
@deepak1556:
```
    // overwrite new NODE_OPTIONS without unsupported variables
    if (new_options != options)
      env->SetVar("NODE_OPTIONS", new_options);
```
needs that i believe?

Reply:
That check seems unnecessary, whats the harm in setting the options irrespective of the check.

Reply:
ah ok! will do 🙇‍♀